From 47a484ba63811f6d4642e702a5c27b77001165e1 Mon Sep 17 00:00:00 2001
From: Leo <thinkabit.ukim@gmail.com>
Date: Thu, 2 Jan 2020 16:05:29 +0100
Subject: [PATCH] lib_tgvoip.cmake: Match against x86 arches.

This avoids matching against other 32bit arches like armv7/armhf that
doesn't have msse2 but are 32bit and need other 32bit-specific flags to
allow for compilation like -g0.
---
 Telegram/cmake/lib_tgvoip.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Telegram/cmake/lib_tgvoip.cmake b/Telegram/cmake/lib_tgvoip.cmake
index 38fa632a40..2801fbec2d 100644
--- Telegram/cmake/lib_tgvoip.cmake
+++ Telegram/cmake/lib_tgvoip.cmake
@@ -756,7 +756,7 @@ else()
         -Wno-error=sequence-point
         -Wno-error=unused-result
     )
-    if (build_linux32)
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "(i686|i386|x86)")
         target_compile_options(lib_tgvoip PRIVATE -msse2)
     endif()
     target_compile_definitions(lib_tgvoip
